harness/determined/common/api/authentication.py

```diff
@@ -218,6 +218,22 @@ def logout(
         pass
 
 
+def logout_all(master_address: Optional[str], cert: Optional[certs.Cert]) -> None:
+    master_address = master_address or util.get_default_master_address()
+    token_store = TokenStore(master_address)
+
+    users = token_store.get_all_users()
+    # Don't log out of DEFAULT_DETERMINED_USER when it's not specified and not the active user.
```

Comment:
Maybe you or somebody who's more familiar with this feature than I am (@rb-determined-ai, perhaps) can explain this to me, but it seems weird that `logout_all` doesn't log out all.

Reply:
I saw this feature in the [logout()](https://github.com/determined-ai/determined/pull/6591/files/8c4b99cc90caf5e36a2220229395d747c1bc6771#diff-7334dcb330f666bfb69eb6c226cf76cb9b0255050fe05db9d5d817a989284272R200), so I suppose we want to do the same thing in logout_all(). But let me know if you all think it's unnecessary.

Reply:
hm, code in `logout()` is mine, and not even that old (november).

I honestly can only kinda remember what case I was guarding against.  I _think_ I was guarding against the case where you logout() once, and it does what you expect, then (from a non-logged-in state) you logout() again and it would log out the `determined` user who wasn't even an active user, because of some of the defaulting logic in other places.

Honestly I hate this part of the code.  And I _hate_ the default user, it introduces so much complexity.

Anyway, I agree with @maxrussell that logout_all shouldn't have the same protection.  It should just log out all sessions associated with this master.